src/test/java/api/subscribers/CloseDeclaredLostLoanWhenLostItemFeesAreClosedApiTests.java

```diff
@@ -0,0 +1,141 @@
+
+  @Before
+  public void createLoanAndDeclareItemLost() {
+    useLostItemPolicy(lostItemFeePoliciesFixture.chargeFee().getId());
+
+    item = itemsFixture.basedUponSmallAngryPlanet();
+    loan = checkOutFixture.checkOutByBarcode(item, usersFixture.jessica());
+
+    declareLostFixtures.declareItemLost(new DeclareItemLostRequestBuilder()
+      .forLoanId(loan.getId()));
+  }
+
+  @Test
+  public void shouldCloseLoanWhenAllFeesClosed() {
+    feeFineAccountFixture.payLostItemFee(loan.getId());
+    feeFineAccountFixture.payLostItemProcessingFee(loan.getId());
+
+    eventSubscribersFixture.publishLoanRelatedFeeFineClosedEvent(loan.getId());
+
+    assertThat(loansFixture.getLoanById(loan.getId()).getJson(), isClosed());
+    assertThat(itemsClient.getById(item.getId()).getJson(), isLostAndPaid());
+  }
+
+  @Test
+  public void shouldNotCloseLoanWhenProcessingFeeIsNotClosed() {
+    feeFineAccountFixture.payLostItemFee(loan.getId());
+
+    eventSubscribersFixture.publishLoanRelatedFeeFineClosedEvent(loan.getId());
+
+    assertThat(loansFixture.getLoanById(loan.getId()).getJson(), isOpen());
+    assertThat(itemsClient.getById(item.getId()).getJson(), isDeclaredLost());
+  }
+
+  @Test
+  public void shouldNotCloseLoanIfItemFeeIsNotClosed() {
+    feeFineAccountFixture.payLostItemProcessingFee(loan.getId());
+
+    eventSubscribersFixture.publishLoanRelatedFeeFineClosedEvent(loan.getId());
+
+    assertThat(loansFixture.getLoanById(loan.getId()).getJson(), isOpen());
+    assertThat(itemsClient.getById(item.getId()).getJson(), isDeclaredLost());
+  }
+
+  @Test
+  public void shouldNotCloseLoanIfActualCostIsUsed() {
+    useLostItemPolicy(lostItemFeePoliciesFixture.create(
+      lostItemFeePoliciesFixture.facultyStandardPolicy()
+        .chargeProcessingFee(10.00)
+        .withActualCost(10.0)).getId());
+
+    feeFineAccountFixture.payLostItemProcessingFee(loan.getId());
+
+    eventSubscribersFixture.publishLoanRelatedFeeFineClosedEvent(loan.getId());
+
+    assertThat(loansFixture.getLoanById(loan.getId()).getJson(), isOpen());
+    assertThat(itemsClient.getById(item.getId()).getJson(), isDeclaredLost());
+  }
+
+  @Test
+  public void shouldNotCloseCheckedOutLoan() {
+    item = itemsFixture.basedUponNod();
+    loan = checkOutFixture.checkOutByBarcode(item, usersFixture.jessica());
+
+    eventSubscribersFixture.publishLoanRelatedFeeFineClosedEvent(loan.getId());
+
+    assertThat(loansFixture.getLoanById(loan.getId()).getJson(), isOpen());
+    assertThat(itemsClient.getById(item.getId()).getJson(), isCheckedOut());
+  }
+
+  @Test
+  public void shouldNotCloseLoanIfLoanIdDoesNotPresent() {
+    final Response response = eventSubscribersFixture
+      .attemptPublishLoanRelatedFeeFineClosedEvent(null, UUID.randomUUID());
+
+    assertThat(response.getJson(), hasErrorWith(allOf(
+      hasMessage("Loan id is required"),
+      hasNullParameter("loanId")
+    )));
+  }
+
+  @Test
+  public void shouldNotCloseNonExistentLoan() {
```

Comment:
```suggestion
  public void cannotProcessMessageIfLoanDoesNotExist() {
```

Reply:
Logic now modified and doesn't fail but just logs cause and returns 204 always.

Reply:
Is there any way to demonstrate that or should this test be removed?

Reply:
I've updated the test it verifies that 204 is returned regardless wrong data provided.